### PR TITLE
Wait for querynode data sync

### LINKED ISSUE #1557: Transaction status modal
Update the transaction status modals to use a new design:

*   [x] [Waiting](https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-Design?node-id=5948%3A165959) for the extension
*   [x] Transaction [canceled](https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-Design?node-id=5948%3A165843)
*   [x] [Pending](https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-Design?node-id=5948%3A165117) transaction
*   [ ] [Finalizing](https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-Design?node-id=5948%3A165264) transaction
*   [ ] Transaction [failed](https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-Design?node-id=5948%3A165264)
*   [ ] Processing transaction (see #1559)
*   [ ] Transaction [succeeded](https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-Design?node-id=5948%3A165567)
* [x] Extension sign modal closed (signing rejected)

### LINKED ISSUE #1559: Transaction should wait for query node
Since the Pioneer 2 fetches the current state from the query-node we should wait with displaying that transaction succeded to the moment when it is processed by the query-node.

In order to do that, we must have a general mechanism that registers a GraphQL query that checks if the transaction's block has been mined.

See the Atlas for details on how to implement this.

### PR DESCRIPTION
**Pull request by RafalPloszka**
Closes #1559
Closes #1557

### CODE REVIEW

File: packages/ui/src/common/hooks/useSignAndSendTransaction.ts [hunk 1]
```diff
@@ -14,6 +14,11 @@ interface Params {
   service: ActorRef<any>
 }
 
+// Transactions which emit events handled by QueryNode use useSignAndSendQueryNodeTransaction hook
```
Comment (RafalPloszka) [inline packages/ui/src/common/hooks/useSignAndSendTransaction.ts:17]: I chose which transactions emit events handled by QueryNode (and therefore need to use useSignAndSendQueryNodeTransaction hook) based on:
https://github.com/Joystream/joystream/blob/olympia/query-node/manifest.yml

File: packages/ui/src/common/hooks/useSignAndSendTransaction.ts [hunk 2]
```diff
@@ -14,6 +14,11 @@ interface Params {
...
+// that waits for QueryNode confirmation on PROCESSING stage.
+// Other transactions use simplified hook useSignAndSendTransaction which automatically switch
+// from state FINALIZING to PROCESSING.
+
```
Comment (RafalPloszka) [inline packages/ui/src/common/hooks/useSignAndSendTransaction.ts]: I chose which transactions emit events handled by QueryNode (and therefore need to use useSignAndSendQueryNodeTransaction hook) based on:
https://github.com/Joystream/joystream/blob/olympia/query-node/manifest.yml

File: packages/ui/src/common/hooks/useSignAndSendTransaction.ts [hunk 3]
```diff
@@ -37,31 +40,32 @@ const observeTransaction = (
...
   }
 
-  const errorHandler = () => send({ type: 'ERROR', events: [] })
+  const errorHandler = () => send({ type: 'CANCELED', events: [] })
 
-  transaction.subscribe(statusCallback, errorHandler)
+  transaction.subscribe({
+    next: statusCallback,
+    error: errorHandler,
+  })
```
Comment (RafalPloszka) [inline packages/ui/src/common/hooks/useSignAndSendTransaction.ts]: Update deprecated signature of the subscribe method:
https://rxjs.dev/deprecations/subscribe-arguments

File: packages/ui/src/common/hooks/useSignAndSendTransaction.ts [hunk 4]
```diff
@@ -1,91 +1,64 @@
...
-  send: Sender<any>,
-  fee: BN,
-  setPending: (b: boolean) => void
-) => {
-  const statusCallback = (result: ISubmittableResult) => {
-    const { status, events } = result
+// Transactions which emit events handled by QueryNode use useSignAndSendQueryNodeTransaction hook
+// that waits for QueryNode confirmation on PROCESSING stage.
+// Other transactions use simplified hook useSignAndSendTransaction which automatically switch
+// from PROCESSING state to SUCCESS/ERROR.
```
Comment (RafalPloszka) [inline packages/ui/src/common/hooks/useSignAndSendTransaction.ts:20]: I chose which transactions emit events handled by QueryNode (and therefore need to use useSignAndSendQueryNodeTransaction hook) based on:
https://github.com/Joystream/joystream/blob/olympia/query-node/manifest.yml

**Review by jodator — COMMENTED** [review]
Overall tremendous work 🚀  Since other tasks depends on this I'll create follow-up tickets if I find anything.

**Pull request merged**